src/rust/ide/view/graph-editor/src/component/node/output/port.rs

```diff
@@ -0,0 +1,458 @@
+    ///
+    /// The crop plane is a `HalfPlane` that is perpendicular to the border of the shape and can be
+    /// used to crop the shape at the specified port index.
+    fn compute_crop_plane
+    ( index           : &Var<f32>
+    , port_num        : &Var<f32>
+    , width           : &Var<f32>
+    , corner_radius   : &Var<f32>
+    , position_offset : &Var<f32>
+    ) -> AnyShape {
+        let corner_circumference  = corner_radius * 2.0 * PI;
+        let corner_segment_length = &corner_circumference * 0.25;
+        let center_segment_length = width - corner_radius * 2.0;
+        let shape_border_length   = &center_segment_length + &corner_segment_length * 2.0;
+
+        let position_relative = index / port_num;
+        let crop_segment_pos  = &position_relative * &shape_border_length + position_offset;
+
+        let crop_plane_pos_relative = calculate_crop_plane_position_relative_to_center_segment
+            (&shape_border_length,&corner_segment_length,&crop_segment_pos);
+        let crop_plane_pos          = crop_plane_pos_relative * &center_segment_length;
+        let crop_plane_pos          = crop_plane_pos + corner_radius;
+
+        let plane_rotation_angle = compute_border_perpendicular_angle
+            (&shape_border_length,&corner_segment_length,&crop_segment_pos);
+        let plane_shape_offset  = Var::<Pixels>::from(&crop_plane_pos - width * 0.5);
+
+        let crop_shape = HalfPlane();
+        let crop_shape = crop_shape.rotate(plane_rotation_angle);
+        let crop_shape = crop_shape.translate_x(plane_shape_offset);
+
+        crop_shape.into()
+    }
+
+    ensogl::define_shape_system! {
+        ( style           : Style
+        , size_multiplier : f32
+        , index           : f32
+        , opacity         : f32
+        , port_count      : f32
+        , padding_left    : f32
+        , padding_right   : f32
+        , color_rgb       : Vector3<f32>
+        ) {
+            let overall_width  = Var::<Pixels>::from("input_size.x");
+            let overall_height = Var::<Pixels>::from("input_size.y");
+            let ports          = AllPortsShape::new(&overall_width,&overall_height,&size_multiplier);
+
+            let inner_radius = Var::<f32>::from(ports.inner_radius);
+            let inner_width  = Var::<f32>::from(ports.inner_width);
+            let next_index   = &index + &Var::<f32>::from(1.0);
+
+            let left_shape_crop = compute_crop_plane
+                (&index,&port_count,&inner_width,&inner_radius,&0.0.into());
+            let right_shape_crop = compute_crop_plane
+                (&next_index,&port_count,&inner_width,&inner_radius,&0.0.into());
+
+            let hover_area = ports.hover.difference(&left_shape_crop);
+            let hover_area = hover_area.intersection(&right_shape_crop);
+            // let hover_area = hover_area.fill(color::Rgba::new(1.0,0.0,0.0,0.3));
+            let hover_area = hover_area.fill(color::Rgba::almost_transparent());
```

Comment:
I believe there was a constant was independent of the colours defined to be used on these cases: `HOVER_COLOR` in  
`src/rust/ensogl/lib/core/src/display/shape/constants.rs`